charts/ai-software-templates/chatbot/templates/application-gitops-job.yaml

```diff
@@ -12,7 +12,118 @@ spec:
+                 tar cf - ./* | (cd ../../../../"$APP_NAME"/.tekton || exit; tar xf -)
+                 cd ../../../../"$APP_NAME" || exit
+              }
+
+              cd /tmp || exit
+              wget https://github.com/cli/cli/releases/download/v"$GH_VERSION"/gh_"$GH_VERSION"_linux_amd64.tar.gz || exit
+              tar xzf gh_"$GH_VERSION"_linux_amd64.tar.gz
+              GH_PATH=$(pwd)/gh_"$GH_VERSION"_linux_amd64/bin
+              export PATH="$GH_PATH":"$PATH"
+              export SOURCE_REPO_APP_CONTENT_PATH="chatbot"
+              export SOURCE_TEKTON_REPO_CONTENT_PATH="pac/pipelineRuns"
+              export TEKTON_FILE_APP_NAME_REPLACEMENT="application_name_replace"
+              export TEKTON_FILE_APP_NAMESPACE_REPLACEMENT="application_namespace_replace"
+              export TEKTON_FILE_QUAY_ACCOUNT_REPLACEMENT="quay_account_replace"
+              echo APP_NAME is "$APP_NAME"
+              echo APP_NAMESPACE is "$APP_NAMESPACE"
+              echo GITHUB_ORG_NAME is "$GITHUB_ORG_NAME"
+              echo GITHUB_SOURCE_REPO is "$GITHUB_SOURCE_REPO"
+              echo GITHUB_TETKON_SOURCE_REPO is "$GITHUB_TETKON_SOURCE_REPO"
+              echo GITHUB_DEFAULT_BRANCH is "$GITHUB_DEFAULT_BRANCH"
+              echo QUAY_ACCOUNT_NAME is "$QUAY_ACCOUNT_NAME"
+
+              export GH_TOKEN=$GITHUB_TOKEN
+              export GH_REPO="github.com/$GITHUB_ORG_NAME/$APP_NAME"
+
+              export HOME=/tmp
+
+              gh auth login -p https
+              gh auth setup-git --hostname=github.com
+
+              gh repo view "$GH_REPO" --json id 2> /dev/null
+
+              return_code=$?
+
+              if [ $return_code == 0 ]; then
+                echo Repo "$GH_REPO" already exists, checking if contents are OK
+                setup_data
+                cd $HOME || exit
+                gh repo clone "$GH_REPO" existing -- --no-tags --branch="$GITHUB_DEFAULT_BRANCH" --single-branch || exit
+                same=$(diff existing/Containerfile "$APP_NAME"/Containerfile)
+                return_code=$?
+                if [ $return_code != 0 ]; then
+                  echo "$same"
+                  exit 1
+                fi
+                same=$(diff existing/chatbot_ui.py "$APP_NAME"/chatbot_ui.py)
+                return_code=$?
+                if [ $return_code != 0 ]; then
+                  echo "$same"
+                  exit 1
+                fi
+                same=$(diff existing/requirements.txt "$APP_NAME"/requirements.txt)
+                return_code=$?
+                if [ $return_code != 0 ]; then
+                  echo "$same"
+                  exit 1
+                fi
+                same=$(diff existing/.tekton/docker-push.yaml "$APP_NAME"/.tekton/docker-push.yaml)
+                return_code=$?
+                if [ $return_code != 0 ]; then
+                  echo "$same"
+                  exit 1
+                fi
```

Comment:
It looks like all of these checks are the same just with different files, I think you could do a list of the files and loop over them `similar` to this (`warning this code is untested`) to simplify things and make it easier to add more files in the future.
```suggestion
FILES=(
"Containerfile"
"chatbot_ui.py"
"requirements.txt"
".tekton/docker-push.yaml"
)

# Iterate over the FILES array
for FILE in "${FILES[@]}"
do
  same=$(diff existing/"$FILE" "$APP_NAME"/"$FILE")
  return_code=$?
  if [ $return_code != 0 ]; then
    echo "$same"
    exit 1
  fi
done
```

Reply:
yeah I'm not good with for loops in bash .... I'll give yours a try

Reply:
very close .... took some iterating but 
```
                FILES=("Containerfile"
                "chatbot_ui.py"
                "requirements.txt"
                ".tekton/docker-push.yaml")

                for FILE in "${FILES[@]}"
                do
                  echo Looking at "${FILE}"
                  same=$(diff existing/"${FILE}" "$APP_NAME"/"${FILE}")
                  return_code=$?
                  if [ $return_code != 0 ]; then
                    echo "$same"
                  fi
                done
```
does the trick